qutip/cy/piqs.pyx

```diff
@@ -0,0 +1,665 @@
+        level systems.
+    """
+    if N % 2 == 0:
+        return 0
+    else:
+        return 0.5
+
+
+def j_vals(N):
+    """
+    Get the valid values of j for given N.
+
+    Parameters
+    ----------
+    N: int
+        The number of two-level systems.
+
+    Returns
+    -------
+    jvals: np.ndarray
+        The j values for given N as a 1D array.
+    """
+    j = np.arange(j_min(N), N/2 + 1, 1)
+    return j
+
+
+def m_vals(j):
+    """
+    Get all the possible values of m or m1 for given j.
+
+    Parameters
+    ----------
+    N: int
+        The number of two-level systems.
+
+    Returns
+    -------
+    mvals: np.ndarray
+        The m values for given j as a 1D array.
+    """
+    return np.arange(-j, j + 1, 1)
+
+
+def get_index(N, j, m, m1, blocks):
+    """
+    Get the index in the density matrix for this j, m, m1 value.
+
+    Parameters
+    ----------
+    N: int
+        The number of two-level systems.
+
+    j, m, m1: float
+        The j, m, m1 values.
+
+    blocks: np.ndarray
+        An 1D array with the number of cumulative elements at the boundary of
+        each block.
+
+    Returns
+    -------
+    mvals: array
+        The m values for given j.
+    """
+    _k = int(j-m1)
+    _k_prime = int(j-m)
+    block_number = int(N/2 - j)
+    offset = 0
+    if block_number > 0:
+        offset = blocks[block_number-1]
+    i = _k_prime + offset
+    k = _k + offset
+    return (i, k)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef list jmm1_dictionary(int N):
+    """
+    Get the index in the density matrix for this j, m, m1 value.
+
+    The (j, m, m1) values are mapped to the (i, k) index of a block
+    diagonal matrix which has the structure to capture the permutationally
+    symmetric part of the density matrix. For each (j, m, m1) value, first
+    we get the block by using the "j" value and then the addition in the
+    row/column due to the m and m1 is determined. Four dictionaries are
+    returned giving a map from the (j, m, m1) values to (i, k), the inverse
+    map, a flattened map and the inverse of the flattened map.
+    """
+    cdef long i
+    cdef long k
+    cdef dict jmm1_dict = {}
+    cdef dict jmm1_inv = {}
+    cdef dict jmm1_flat = {}
+    cdef dict jmm1_flat_inv = {}
+    cdef int l
+    cdef int nds = _num_dicke_states(N)
+    cdef list blocks = get_blocks(N)
+
+    jvalues = j_vals(N)
+    for j in jvalues:
+        mvalues = m_vals(j)
+        for m in mvalues:
+            for m1 in mvalues:
+                i, k = get_index(N, j, m, m1, blocks)
+                jmm1_dict[(i, k)] = (j, m, m1)
+                jmm1_inv[(j, m, m1)] = (i, k)
+                l = nds * i+k
+                jmm1_flat[l] = (j, m, m1)
+                jmm1_flat_inv[(j, m, m1)] = l
+    return [jmm1_dict, jmm1_inv, jmm1_flat, jmm1_flat_inv]
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef class Dicke(object):
+    """
+    The Dicke States class.
+
+    Parameters
```

Comment:
These seem like descriptions for the Attributes. You could just change the heading to `Attributes`
You should also describe the `Parameters` in a separate section here. Those would be the parameters of the __init__ method. Could just be copy-paste, but best to check.
Some don't seem to be attributes or parameters of this class